File: problems/2145-count-the-hidden-sequences/solution.py
# Link: https://leetcode.cn/problems/count-the-hidden-sequences/
# Level: Medium
# Date: 2025-04-21

class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        num = 0
        min_num = 0
        max_num = 0
        for d in differences:
            num += d
            min_num = min(min_num, num)
            max_num = max(max_num, num)
            if max_num - min_num > upper - lower:
                return 0
        return (upper - lower) - (max_num - min_num) + 1

File: problems/2145-count-the-hidden-sequences/solution.cpp
// Link: https://leetcode.cn/problems/count-the-hidden-sequences/
// Level: Medium
// Date: 2025-04-21

class Solution {
    public:
        int numberOfArrays(vector<int>& differences, int lower, int upper) {
            int num = 0;
            int min_num = 0;
            int max_num = 0;
            for (int d : differences) {
                num += d;
                min_num = min(num, min_num);
                max_num = max(num, max_num);
                if (max_num - min_num > upper - lower) {
                    return 0;
                }
            }
            int res = (upper - lower) - (max_num - min_num) + 1;
            return res;
        }
};
